types/node/index.d.ts

```diff
@@ -756,20 +756,23 @@ declare module "http" {
         write(str: string, encoding?: string, cb?: Function): boolean;
         write(str: string, encoding?: string, fd?: string): boolean;
 
-        writeContinue(): void;
-        writeHead(statusCode: number, reasonPhrase?: string, headers?: any): void;
-        writeHead(statusCode: number, headers?: any): void;
-        statusCode: number;
-        statusMessage: string;
+        addTrailers(headers: any): void;
+        finished: boolean;
+        getHeader(name: string): string;
+        getHeaderNames(): string[]
+        getHeaders(): any
```

Comment:
To my understanding the use of Typescript interfaces is perfectly fine here. An interface like `ServerResponseHeaders` doesn't imply that you get something that inherits from Object (as you don't wrote that it extends Object) - it just describes which properties are present and of which type they are.